integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -148,7 +154,11 @@ def count_documents(self) -> int:
 
         :returns: how many documents are present in the document store.
         """
-        return self.collection.count()
+        self._ensure_initialized()
+        if self._collection is None:
+            msg = "Collection is not initialized"
+            raise ValueError(msg)
```

Comment:
Both the collection and client variables should be initialized after the call to `_ensure_initialized` , so there's no reason to raise. We can replace that with `assert self._collection is not None` instead (same with the other variable wherever it needs to be directly accessed).